workers/go/kitchensink/kitchen_sink.go

```diff
@@ -96,18 +98,24 @@ func handleActionSet(
 
 func handleAction(
 	ctx workflow.Context,
+	workflowState **kitchensink.WorkflowState,
 	action *kitchensink.Action,
-) (returnValue *common.Payload, err error) {
+) (*common.Payload, error) {
 	if rr := action.GetReturnResult(); rr != nil {
 		return rr.ReturnThis, nil
 	} else if re := action.GetReturnError(); re != nil {
 		return nil, temporal.NewApplicationError(re.Failure.Message, "")
 	} else if can := action.GetContinueAsNew(); can != nil {
 		return nil, workflow.NewContinueAsNewError(ctx, KitchenSinkWorkflow, can.Arguments)
 	} else if timer := action.GetTimer(); timer != nil {
-		if err := workflow.Sleep(ctx, time.Duration(timer.Milliseconds)*time.Millisecond); err != nil {
-			return nil, err
-		}
+		return nil, withAwaitableChoice(ctx, func(ctx workflow.Context) workflow.Future {
+			fut, setter := workflow.NewFuture(ctx)
+			workflow.Go(ctx, func(ctx workflow.Context) {
```

Comment:
Why not just use a timer?

Reply:
Instead of what? The timer needs to be cancellable somehow and the only way to do that in conjunction with the different awaitable choices is to put it in another routine

Reply:
Instead of this sleep in a goroutine, it is very confusing to read when you can just create a timer and treat it's  future like an activity future.

Reply:
The timer doesn't have a future is the problem